stan/math/gpu/opencl_context.hpp

```diff
@@ -328,12 +336,12 @@ class opencl_context {
               << device.getInfo<CL_DEVICE_MAX_MEM_ALLOC_SIZE>() << "\n";
           msg << "\t\tDevice Local Memory: "
               << device.getInfo<CL_DEVICE_LOCAL_MEM_SIZE>() << "\n";
-          msg << "\t\tDevice Available: "
-              << device.getInfo<CL_DEVICE_AVAILABLE>() << "\n";
+          msg << "\t\tDevice Available: " <<
+              device.getInfo<CL_DEVICE_AVAILABLE>() << "\n";
         }
       } catch (const cl::Error& e) {
-        msg << "\tno GPU devices in the specified platform"
-            << "\n";
+        msg << "\tno GPU devices in the platform with ID "
+              << platform_id << "\n";
```

Comment:
This is an example output listing all the platforms on a system. The first 2 have no GPU devices and the last one has 1 GPU device.

`Number of Platforms: 3
Platform ID: 0
Platform Name: Experimental OpenCL 2.1 CPU Only Platform
Platform Vendor: Intel(R) Corporation
	no GPU devices in the platform with ID 1
Platform ID: 1
Platform Name: Intel(R) OpenCL
Platform Vendor: Intel(R) Corporation
	no GPU devices in the platform with ID 2
Platform ID: 2
Platform Name: NVIDIA CUDA
Platform Vendor: NVIDIA Corporation
	Device 0: 
		Device Name: Graphics Device
		Device Type: 4
		Device Vendor: NVIDIA Corporation
		Device Max Compute Units: 30
		Device Global Memory: 12780699648
		Device Max Clock Frequency: 1582
		Device Max Allocateable Memory: 3195174912
		Device Local Memory: 49152
		Device Available: 1
`